staging/cse/windows/provisioningscripts/kubeletstart.ps1

```diff
@@ -66,6 +76,22 @@ if ($global:ContainerRuntime -eq "containerd") {
     $KubeletArgList += @("--container-runtime=remote", "--container-runtime-endpoint=npipe://./pipe/containerd-containerd")
 }
 
+# Update args to get both IPs from NIC using the default route 0.0.0.0/0 if IsDualStackEnabled
+# and the --node-ip argument is not present in the existing kubelet arg list.
+if ($global:IsDualStackEnabled -and !(Test-NodeIPArgExists $KubeletArgList)) {
+    $defaultRouteIfIndex = (Get-NetRoute -DestinationPrefix 0.0.0.0/0).ifIndex
+
+    # use addresses from dhcp and sort by family so IPv4 is always first
+    $ipAddrs = (Get-NetIPAddress -InterfaceIndex $defaultRouteIfIndex | ? { $_.SuffixOrigin -eq "Dhcp" } | Sort-Object AddressFamily).IPAddress
+
+    if ($ipAddrs.Length -eq 2) {
+        $KubeletArgList += @("--node-ip=$($ipAddrs -join ",")")
+    } else {
+        # error instead?
+        Write-Host "Expected exactly 2 IPs for dual-stack, got $($ipAddrs.Length). IP addresses from interface ${defaultRouteIfIndex}: $($ipAddrs -join " ")"
```

Comment:
I think that we should set this parameter in `c:\k\kubeclusterconfig.json` in node provisioning so we can return a new error code for this in cse exitcode. I think that you can reference the function `Write-KubeClusterConfig` in staging/cse/windows/kubernetesfunc.ps1

Reply:
will `Write-KubeClusterConfig` run even on a node restart? if a node gets a new IP between reboots will it write a new config? otherwise it could keep around stale IP data

Reply:
cse will only be run once in provisioning the node. Could you share when the IP will be changed after a reboot? I did not know this before.

Reply:
You can see that currently we also setup the node with the IP in cse

Reply:
also, I am not sure I think moving it to `Write-KubeConifg` is better. We already append additional arguments to the kubelet args here in kubeletstart.ps1 and we aren't doing that in `Write-KubeConfig`. What about just adding validation inside CSE to check `if ($global:IsDualStackEnabled) { expect 2 IPs on the NIC else return CSE error }` but keep the `--node-ip` setup here in kubeletstart?

Reply:
@tyler-lloyd I think that the cse will be run again if the vm is started from deallocation. For IPv4, we always need to get the current host ip in provisioning Windows nodes in https://github.com/Azure/AgentBaker/blob/master/staging/cse/windows/azurecnifunc.ps1#L363

Reply:
sounds good, updated in https://github.com/Azure/AgentBaker/pull/2813/commits/e90990d18d93a6ff92c66b6569e548335dc3e972